rpd_generator/bdl_structure/bdl_commands/system.py

```diff
@@ -1330,6 +1394,73 @@ def populate_air_energy_recovery(self):
                 self.air_energy_recovery_outdoor_airflow
             )
 
+    def populate_fan_operation_during_occupied(self):
+        fan_sch_name = self.keyword_value_pairs.get(BDL_SystemKeywords.FAN_SCHEDULE)
+
+        if (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.RESVVT
+        ):
+            return FanSystemOperationOptions.CYCLING
+
+        elif (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.DOAS
+        ):
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            if not fan_sch:
+                return FanSystemOperationOptions.INTERMITTENT
+            else:
+                return FanSystemOperationOptions.CONTINUOUS
+
+        elif self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE) in [
+            BDL_SystemTypes.PTAC,
+            BDL_SystemTypes.HP,
+            BDL_SystemTypes.UVT,
+        ]:
+            if not self.fan_sys_minimum_outdoor_airflow:
+                return FanSystemOperationOptions.CYCLING
+
+            min_oa_sch_name = self.keyword_value_pairs.get(
+                BDL_SystemKeywords.MIN_AIR_SCH
+            )
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            min_oa_sch = self.rmd.bdl_obj_instances.get(min_oa_sch_name)
+            if not min_oa_sch:
+                return FanSystemOperationOptions.CONTINUOUS
+            if not fan_sch:  # (and min_oa_sch)
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if min_oa_sch_values:
+                    for i in range(len(min_oa_sch_values)):
+                        if min_oa_sch_values[i] == 0:
+                            return FanSystemOperationOptions.CYCLING
```

Comment:
I think I am probably misunderstanding something but this looks like if any value in schedule is 0 it returns cycling. But I think it could legitimately have a 0 value during unoccupied hours and still operate continuously during occupied hours (schedule values of 1 during occupied hours).

Reply:
Ok I think this makes sense - essentially if there is no fan schedule but there is a min OA schedule then we translate the "occupied" assumption to the min OA schedule instead of the fan schedule

Reply:
Although I think if you really wanted to have an unoccupied mode you would use the fan schedule so I am second guessing if this makes sense to do. Part of me thinks if fan schedule is not assigned then it should be treated as always occupied and so any 0 in the min OA schedule would mean intermittent.

Reply:
hmm for these system types I am not sure I agree since they operate as cycling with load except when OA is provided so a project could esstentially use the min-OA schedule to force the fan to operate continuously during occ hours and then cycle with load during unocc hours.

Reply:
that makes sense, thanks!